src/foam/dao/FixedSizeDAO.js

```diff
@@ -51,11 +52,17 @@ foam.CLASS({
         Count count = new Count();
         count = (Count) this.getDelegate().select(count);
         if ( count.getValue() > getSize() + getSize() * getPurgePercent() / 100 ) {
-          this.getDelegate()
+          DAO delegate = this.getDelegate()
             .where(getPredicate())
-            .orderBy(getComparator())
-            .skip(getSize())
-            .removeAll();
+            .skip(getSize());
+
+          if ( getComparator() != null ) {
```

Comment:
It is odd that of all the places we take comparators, this one takes an array instead. 
I've added a couple convenience methods to make it easier to create compound comparators:
commit 046f4371e02ec48ea9f58ae42722b7b937671b92
Author: Kevin Greer <kgrgreer@gmail.com>
Date:   Wed Oct 28 14:44:24 2020 -0400

    Add THEN_BY and COMPOUND to MLang.java.